File: 1_Java_Examples/1_ConferenceAttendees_SRP/PriceCalculator.java
public class PriceCalculator {
	
	private static int REGULAR_PRICE = 1000;
	
	public int calculatePrice(Attendee attendee) {
		switch (attendee.getType()) {
			case Speaker: return REGULAR_PRICE / 2;
			case Organizer: return 0;
			case Participant: return REGULAR_PRICE;
			default: throw new IllegalStateException();
		}
	}
	
}

File: 3_Scala_Examples/1_ConferenceAttendees_SRP/PriceCalculator.scala
object PriceCalculator {
	def calculatePrice(attendee:Attendee):Int = {
          val REGULAR_PRICE = 1000
	  attendee.attendeeType match {
	    case AttendeeType.Participant => return REGULAR_PRICE
	    case AttendeeType.Speaker => return REGULAR_PRICE / 2
	    case AttendeeType.Organizer => return 0
	    case _ => throw new IllegalStateException()
	  }
	}
}
